components/scream/src/physics/p3/p3_functions_update_prognostics_impl.hpp

```diff
@@ -84,6 +84,40 @@ ::update_prognostic_ice(const Spack& qcheti,      const Spack& qccol,  const Spa
 
 }
 
+  template<typename S, typename D>
+  KOKKOS_FUNCTION
+  void Functions<S,D>
+  ::update_prognostic_liquid(const Spack& qcacc, const Spack& ncacc,
+			     const Spack& qcaut,const Spack& ncautc, const Spack& qcnuc, const Spack& ncautr,
+			     const Spack& ncslf, const Spack& qrevp, const Spack& nrevp, const Spack& nrslf,
+			     const bool log_predictNc, const Spack& inv_rho, const Spack& exner, const Spack& xxlv,
+			     const Scalar dt, Spack& th, Spack& qv, Spack& qc, Spack& nc, Spack& qr, Spack& nr)
+  {
+
+    qc = qc + (qcnuc - qcacc - qcaut) * dt;
+    qr = qr + (qcacc + qcaut - qrevp) * dt;
+
+    if(log_predictNc) {
+      nc = nc + (ncslf -ncacc - ncautc) * dt;
+    }
+    else{
+      constexpr Scalar NCCNST = C::NCCNST;
+      nc = NCCNST * inv_rho;
+    }
+    constexpr int IPARAM = C::IPARAM;
+    if (IPARAM == 1 || IPARAM == 2){
+      nr = nr + (0.5 * ncautc - nrslf - nrevp) * dt;
```

Comment:
This 0.5 here will cause problems in single precisions. You need to use the scream-precision macro.
```c++
nr = nr + (sp(0.5) * ncautc - nrslf - nrevp) * dt;
```